playbooks/secrets.yml

```diff
@@ -0,0 +1,114 @@
+---
+- hosts: localhost
+  gather_facts: false
+  vars:
+    base_folder: "{{ playbook_dir }}/.."
+    secrets_file: "{{ base_folder }}/vars/secrets.yml"
+  tasks:
+    - name: Ensure no secrets are still set has group_vars/hostvars
+      assert:
+        that:
+          - hostvars[item].repo_db_password is not defined
+          - hostvars[item].sync_db_password is not defined
+          - hostvars[item].reposearch_shared_secret is not defined
+        quiet: true
+      loop: "{{ groups['database'] | default([]) + groups['repository'] | default([]) }}"
```

Comment:
I think this really needs to be a tasks assigned to all hosts but external. 
* avoid maintaining the loop items
* provides wider control on vars leaking secrets on unexpected hosts